src/NeoSharp.BinarySerialization/BinaryDeserializer.cs

```diff
@@ -0,0 +1,190 @@
+using NeoSharp.BinarySerialization.Cache;
+using NeoSharp.BinarySerialization.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace NeoSharp.BinarySerialization
+{
+    public class BinaryDeserializer : IBinaryDeserializer
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public BinaryDeserializer() { }
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="asms">Assemblies</param>
+        public BinaryDeserializer(params Assembly[] asms)
+        {
+            BinarySerializerCache.CacheTypesOf(asms);
+        }
+
+        /// <summary>
+        /// Cache types (call me if you load a new plugin or module)
+        /// </summary>
+        /// <param name="asms">Assemblies</param>
+        public static void CacheTypesOf(params Assembly[] asms)
```

Comment:
I see what you mean but I think about introduction if instance method as AddTypes/RegisterTypes